tuxemon/mod_manager.py

```diff
@@ -0,0 +1,190 @@
+
+        self.write_package_to_list(outfolder, name)
+
+        if not dont_extract:
+            self.extract(filename, outfolder)
+            with open(f"{outfolder}/meta.json", "w") as metafile:
+                meta = self.get_package_info(name, repo)
+                metafile.write(json.dumps(meta, indent=4, sort_keys=False))
+
+        if install_deps:
+            # This function calls download_package, might cause issues
+            self.install_dependencies(name, release, repo, dont_extract=dont_extract)
+
+    def install_dependencies(self, name, release, repo, dont_extract=False, symlink=True):
+        """
+        Same as the download_package(), but it includes dependency installing.
+        When symlink is True, dependency's files will be linked.
+        """
+        # Get info
+        meta = self.get_package_info(name, repo)
+        if "dependencies" in meta:
+            dependencies = meta["dependencies"]
+
+            for pack in dependencies:
+                self.download_package(pack, release, repo, dont_extract=dont_extract, install_deps=False)
+
+                # Symlink deps
+                mainfolder = os.path.join(paths.BASEDIR, "mods", name)
+                depfolder = os.path.join(paths.BASEDIR, "mods", pack)
+                symlink_missing(mainfolder, depfolder)
+        else: pass
+
+    def extract(self, file, outfolder):
+        """Extracts the specified zip archive to the mods directory"""
+        with zipfile.ZipFile(file) as zip_:
+            zip_.extractall(path=os.path.join(paths.BASEDIR, "mods", outfolder))
+
+    def get_package_info(self, name, repo):
+        """Get specified package info. Always downloads the info from the server."""
+        with urllib.request.urlopen(repo + f"/api/packages/{name}") as packages:
+            return json.loads(packages.read().decode("UTF-8"))
+
+    def get_package_repo(self, name):
+        """Reads the origin of an package.
+        Returns None, if key 'mods' doesn't exist"""
+        for i in self.packages:
+            if i["name"] == name:
+                return i["repo"]
+                """
+                if "repo" in i: return i["repo"]
+                else: return None
+                """
+            else: continue
+
+    def write_package_to_list(self, path_to_folder, name):
+        """Writes specified package to the package list"""
+        # Write the absolute path to the list
+        with open(paths.USER_GAME_DATA_DIR + "/package.list", "w+") as file:
+            if not len(file.read()) == 0:
+                before = json.loads(file.read())
+            else:
+                before = {}
+
+            to_append = {name: path_to_folder}
+            after = {**before, **to_append}
+            file.write(
+                json.dumps(after, indent=4)
+            )
+
+    def read_package_from_list(self, name):
+        """Reads path of the specified mod"""
+        #try:
```

Comment:
please avoid leaving commented code, if possible.  there is a bit in Tuxemon already, for legacy code, but we should avoid adding new commented code lines.